lib/mongo/server/connection_pool.rb

```diff
@@ -520,16 +563,36 @@ def check_in(connection)
         check_invariants
       end
 
+      def pause(options = nil)
+        raise_if_closed!
+
+        check_invariants
+
+        if Lint.enabled? && !@server.unknown?
+          raise Error::LintError, "Attempting to pause pool for server #{@server.summary} which is known"
+        end
+
+        return if paused?
+
+        @lock.synchronize do
+          @ready = false
+        end
+
+        stop_populator
+
+        clear(options)
```

Comment:
I'm not sure why we decided to automatically clear on pause here but that does not look right. I will attempt to repair this.